### PR TITLE
capi: Cleanup in fizzy.h comments

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: include/fizzy/fizzy.h [hunk 1]
```diff
@@ -145,6 +146,8 @@ typedef struct FizzyImportedFunction
 } FizzyImportedFunction;
 
 /// Validate binary module.
+///
+/// @return true if module is valid, false otherwise.
```
Comment (axic) [inline include/fizzy/fizzy.h:150]: Well, even if it is basic, shouldn't the params also be included?

File: include/fizzy/fizzy.h [hunk 2]
```diff
@@ -273,8 +277,9 @@ void fizzy_free_instance(FizzyInstance* instance);
 
 /// Get pointer to module of an instance.
 ///
-/// @note  The returned pointer represents non-owning, "view"-access to the module and must not be
-///        passed to fizzy_free_module.
+/// @return  Pointer to a module.
```
Comment (axic) [inline include/fizzy/fizzy.h:281]: Well, even if it is basic, shouldn't the params also be included?

File: include/fizzy/fizzy.h [hunk 3]
```diff
@@ -19,7 +19,7 @@ typedef struct FizzyInstance FizzyInstance;
 
 /// The data type representing numeric values.
 ///
-/// i64 member is used to represent values of both i32 and i64 type.
+/// `i64` member is used to represent values of both i32 and i64 type.
```
Comment (axic) [inline include/fizzy/fizzy.h:22]:
```suggestion
/// The `i64` member is used to represent values of both i32 and i64 type.
```

File: include/fizzy/fizzy.h (hunk 1 above)
Comment (gumb0) [reply include/fizzy/fizzy.h]: Added.

File: include/fizzy/fizzy.h (hunk 2 above)
Comment (gumb0) [reply include/fizzy/fizzy.h]: Added.

File: include/fizzy/fizzy.h (hunk 3 above)
Comment (gumb0) [reply include/fizzy/fizzy.h]: Added.

File: include/fizzy/fizzy.h [hunk 4]
```diff
@@ -33,10 +33,10 @@ typedef struct FizzyExecutionResult
     /// Whether execution ended with a trap.
     bool trapped;
     /// Whether function returned a value.
-    /// Equals false if trapped equals true.
+    /// Equals false if `trapped` equals true.
```
Comment (chfast) [inline include/fizzy/fizzy.h:36]:
```suggestion
    /// Equals false if #trapped equals true.
```
I'm not sure that the goal is here, but you can reference the class variable like this. See https://www.doxygen.nl/manual/autolink.html.

**Comment by gumb0** [comment]
I change backticks to proper links, as Pawel suggested, please take a look.

File: include/fizzy/fizzy.h [hunk 5]
```diff
@@ -19,7 +19,7 @@ typedef struct FizzyInstance FizzyInstance;
 
 /// The data type representing numeric values.
 ///
-/// i64 member is used to represent values of both i32 and i64 type.
+/// The `i64` member is used to represent values of both i32 and i64 type.
```
Comment (chfast) [inline include/fizzy/fizzy.h:22]:
```suggestion
/// The #i64 member is used to represent values of both i32 and i64 type.
```

File: include/fizzy/fizzy.h [hunk 6]
```diff
@@ -145,23 +146,31 @@ typedef struct FizzyImportedFunction
...
 const FizzyModule* fizzy_parse(const uint8_t* wasm_binary, size_t wasm_binary_size);
 
 /// Free resources associated with the module.
 ///
-/// Should be called unless @p module was passed to fizzy_instantiate.
-/// If passed pointer is NULL, has no effect.
+/// @param  module    Pointer to module. If NULL is passed, function has no effect.
+///
+/// @note
+/// Should be called unless @p module was passed to ::fizzy_instantiate.
```
Comment (chfast) [inline include/fizzy/fizzy.h:167]:
```suggestion
/// Should be called unless @p module was passed to fizzy_instantiate().
```

File: include/fizzy/fizzy.h [hunk 7]
```diff
@@ -208,15 +218,15 @@ bool fizzy_module_has_start_function(const FizzyModule* module);
...
 ///
-/// The instance takes ownership of the module, i.e. fizzy_free_module must not be called on the
+/// The instance takes ownership of the module, i.e. ::fizzy_free_module must not be called on the
 /// module after this call.
-/// For simplicity a module cannot be shared among several instances (calling fizzy_instatiate more
-/// than once with the same module results in undefined behaviour), but after fizzy_instantiate
-/// functions querying module info can still be called with @p module.
+/// For simplicity a module cannot be shared among several instances (calling ::fizzy_instatiate
+/// more than once with the same module results in undefined behaviour), but after
+/// ::fizzy_instantiate functions querying module info can still be called with @p module.
```
Comment (chfast) [inline include/fizzy/fizzy.h:225]:
```suggestion
/// fizzy_instantiate() functions querying module info can still be called with @p module.
```

**Review by chfast — COMMENTED** [review]
I guess some doxygen CI build would be welcomed to see the changes.

File: include/fizzy/fizzy.h (hunk 5 above)
Comment (chfast) [reply include/fizzy/fizzy.h:22]: Works for me provided the `i64` is itself documented.

**Review by axic — APPROVED** [review]
Looks okay to me. The last two commits could be merged. Also is `@p argname` still prefererd or there's some other shorter notation based on this linking feature?

**Comment by gumb0** [comment]
Changed remaining backticks to links, changed `::function_name` links to `function_name()`, squashed last two commits.

**Comment by gumb0** [comment]
> Looks okay to me. The last two commits could be merged. Also is `@p argname` still prefererd or there's some other shorter notation based on this linking feature?

Looks like `@p` is still the way to go, auto-linking feature doesn't support parameters.

**Review by chfast — APPROVED** [review]

**Pull request merged**